acvm-repo/blackbox_solver/src/lib.rs

```diff
@@ -117,15 +117,27 @@ fn verify_secp256k1_ecdsa_signature(
     };
     // Convert the inputs into k256 data structures
 
-    let signature = Signature::try_from(signature.as_slice()).unwrap();
+    let Ok(signature) = Signature::try_from(signature.as_slice()) else {
+        // Signature `r` and `s` are forbidden from being zero.
+        return false;
+    };
 
     let point = EncodedPoint::from_affine_coordinates(
         public_key_x_bytes.into(),
         public_key_y_bytes.into(),
         true,
     );
-    let pubkey = PublicKey::from_encoded_point(&point).unwrap();
 
+    let pubkey = PublicKey::from_encoded_point(&point);
+    let pubkey = if pubkey.is_some().into() {
```

Comment:
It looks like `.into()` is converting a `bool` to a `bool` here. I'd prefer `if let` in general

Reply:
We're not unwrapping an `Option` here but a [`CtOption`](https://docs.rs/subtle/latest/subtle/struct.CtOption.html#method.is_some) to `pubkey.is_some()` isn't a `bool` but `Choice`.

We can't do an `if let` as `CtOption` isn't an enum but a struct.